### ISSUE TITLE
Make it work on windows and mac

### ISSUE THREAD

**Issue by pieper** [MorphoCloud/SlicerMorphoDepot#2]
See discussion here: https://github.com/pieper/MD_E15/issues/7

**Comment by muratmaga** [comment]
Given that Slicer is going to be start from an application icon, we need:

1. Ask the user to provide the path to gh executable
2. And possibly invoke the `gh auth login` inside the MorphoDepot (this will ensure that gh is working. And probably users will find this a bit more convenient).

**Comment by pieper** [comment]
Another option would be to install gh for the user so that we know the path.

Jc packaged up s5cmd as a [pip-installable package](https://pypi.org/project/s5cmd/) this way and it's worked great for the [idc-index project](https://pypi.org/project/idc-index/).  FYI, once there are many scans with complex metadata it could make sense to have something akin to idc-index for searching and downloading.

And yes, I find the `gh auth` process a bit of a chore that may confuse new users but I'm not sure how much we should try to automate since they could change the procedure at any point.  Invoking the command is probably good, but passwords 2fa, and an copying of codes will have to happen manually.

**Comment by muratmaga** [comment]
> Another option would be to install gh for the user so that we know the path.

Does it work on all platforms? 

We can worry less about gh auth if gh is available from get go...

**Comment by muratmaga** [comment]
> Another option would be to install gh for the user so that we know the path.

Yes, that would be great. In MorphoCloud, I will have the gh CLI available from get go. So that will be less of a concern...

**Comment by pieper** [comment]
Yes, it would be available on all platforms.  We can just copy what Jc did, or maybe just ask him to do it since he's already familiar with the details.

As a side comment, Jc and I discussed making a python package called something like 'slicer-runner' that would use pip to get Slicer or launch existing installations.  We could do that also for SlicerMorph users could easily download and run and manage versions using pip.  This would make python, which any computational scientist should have anyway, the key requirement instead of having different directions for all OS platforms.

**Comment by muratmaga** [comment]
I can't seem to create labels on your repo. I wanted to bump this up on the priority, because some people do indeed want to run MorphoDepotReview on their own computers, and most of them have Macs or Windows, not linux.

**Comment by muratmaga** [comment]
We decided to download the github asset and unzip the executable to appropriate place.

**Comment by pieper** [comment]
Here's a prototype that works: https://github.com/MorphoCloud/SlicerMorphoDepot/blob/main/Experiments/gh-installer.py

Needs to be integrated with the GUI.

**Comment by muratmaga** [comment]
Not sure how to run this, execute script from the python console in Slicer?

**Comment by pieper** [comment]
Yes, sorry I was being brief when I said it needed to be integrated.  There are a few steps left to make everything streamlined.

BTW, I realize we probably need the same thing for git itself, right?  Or do you expect people to be installing that manually as a learning experience?  If we do want to do git we can generalize the logic here.  Most windows machines won't have git and on mac the suggested way is to download Xcode, which us huge and mostly not needed.

**Comment by muratmaga** [comment]
I didn't know that GH cli didn't provide the git itself. Is it possible to do everything (including git actions) with GH cli? I mean ultimately it is your call. I guess you can package and install the git the same way. Whichever is easier.

**Comment by pieper** [comment]
No, gh doesn't include git and in general it tries not to duplicate functionality (just makes a few operations easier if you already know they are going to github) but in general you need to install and use git independently and the gitpython package requires the executable to be installed.  So yes, I think we need to follow the same approach and install git if we don't find it in the standard places.

**Comment by pieper** [comment]
Interestingly, git installation is a bit of a mess.  Unlike the `gh` command that comes with handy archives of the executable, `git` seems to only come via package managers or installers.  I'll poke around some more, but we may need to have people manually install git until we figure out something cleaner.

**Comment by muratmaga** [comment]
Yeah, Mac particularly seems problematic. For windows there is this self-decompressing thumbdrive addition, which we can probably use:
https://git-scm.com/downloads/win

**Comment by pieper** [comment]
Still testing on windows

**Comment by muratmaga** [comment]
seems to work at the all-hands meeting
